bin/initoverlayfs-install

```diff
@@ -1,58 +1,114 @@
 #!/bin/bash
 
-set -ex
+# Constants with default value
+DECOMPRESSOR="lz4cat"
+DECOMPRESSOR_DRACUT="--lz4"
 
-extract_initrd_into_initoverlayfs() {
-  DIR_TO_DUMP_INITRAMFS="/var/tmp/initoverlayfs"
-  mkdir -p "$DIR_TO_DUMP_INITRAMFS"
-
-  file_type=$(file /boot/initramfs-$kver.img)
-  decompressor="lz4cat"
-  decompressor_dracut="--lz4"
-  if [[ "$file_type" == *"ASCII cpio archive (SVR4 with no CRC)"* ]]; then
-    decompressor_dracut=""
-    decompressor="zcat"
-  elif [[ "$file_type" == *"regular file, no read permission"* ]] || \
-       [[ "$file_type" == *"gzip"* ]]; then
-    decompressor_dracut=""
-    decompressor="zcat"
-  fi
-
-  if command -v mkfs.erofs; then
-    fstype="erofs"
-    cd "$DIR_TO_DUMP_INITRAMFS"
-    /usr/lib/dracut/skipcpio /boot/initramfs-$kver.img | $decompressor | cpio -ivd
-    cd -
-    rm -f /boot/initoverlayfs-$kver.img
-    mkfs.erofs /boot/initoverlayfs-$kver.img "$DIR_TO_DUMP_INITRAMFS"
-  else
-    echo "Please install mkfs.erofs"
-    exit 0
-
-    fstype="ext4"
-    dd if=/dev/zero of=/boot/initoverlayfs-$kver.img bs=64M count=1
-    dev=$(losetup -fP --show /boot/initoverlayfs-$kver.img)
-    mkfs.$fstype $dev
-    mount $dev "$DIR_TO_DUMP_INITRAMFS"
-    cd "$DIR_TO_DUMP_INITRAMFS"
-    /usr/lib/dracut/skipcpio /boot/initramfs-$kver.img | zstd -d --stdout | cpio -ivd
-    sync
-    cd -
-    while ! umount "$DIR_TO_DUMP_INITRAMFS"; do
+INITOVERLAYFS_CONF="/etc/initoverlayfs.conf"
+INITRAMFS_DUMP_DIR="/var/tmp/initoverlayfs"
+
+SUPPORTED_FILESYSTEM=("erofs" "ext4")
+SKIPCPIO_BIN="/usr/lib/dracut/skipcpio"
+
+exec_cmd() {
```

Comment:
Would rather if we just executed commands as normal rather than creating this wrapper to be honest. But we could do this:

```
set -eE
trap 'echo "Error: Command failed with exit code $?"' ERR
```

That way we don't have to call exec_cmd for this behaviour

Reply:
The wrapper is more a cosmetic thing to users. Imagine, having a control what kind of message (ASCII colors) we can display to users.

BTW, my bad, I should explained that before but I was afraid to prepare a huge or complex patch.

For example: 
```
exec_cmd INFO "Executing command xyz" "xyz -ls"
exec_cmd WARN "Foobar ..." "zyx -ls"
exec_cmd DEBUG "foobar " "xyz -ls"
```
So users would see:
```
./initoverlayfs --verbose debug
[ INFO     ] Executing xyz -ls...
[ WARN   ] Warning message...
[ DEBUG ]  Debug message...
```

However, I am fine with either way.

Reply:
I will prepare a patch with that removed for now.